src/fuzz/protobufs/utils/spanner_emulator_ddl_statement_proto_to_string.cc

```diff
@@ -0,0 +1,188 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "src/fuzz/protobufs/create_table.pb.h"
+#include "src/fuzz/protobufs/spanner_ddl.pb.h"
+
+#include <google/protobuf/repeated_field.h>
+
+#include <cmath>
+#include <string>
+#include "absl/strings/substitute.h"
+#include "absl/strings/str_cat.h"
+
+using spanner_ddl::SpannerDDLStatement;
+using spanner_ddl::CreateTable;
+using spanner_ddl::Column;
+using spanner_ddl::ColumnDataInfo;
+
+// forward declarations
+std::string toString(SpannerDDLStatement statement);
+std::string toString(CreateTable createTable);
+std::string toString(const google::protobuf::RepeatedPtrField<Column> columns);
+std::string toString(Column column);
+std::string toString(ColumnDataInfo columnDataInfo);
+std::string toString(ColumnDataInfo::ScalarType type, 
+    int length, ColumnDataInfo::LengthType lengthType);
+std::string isColumnNotNullToString(bool isNotNull);
+std::string columnOptionsToString(bool allowCommitTimestamps);
+std::string toPrimaryKeys(const google::protobuf::RepeatedPtrField<Column>& columns);
+std::string columnToPrimaryKey(Column column);
+std::string toString(Column::Orientation orientation);
+
+// Transforms any Emulator DDL statement into a syntactically valid string
+std::string toString(SpannerDDLStatement statement) {
+    using statementType = SpannerDDLStatement::DDLStatementCase;
+    switch (statement.DDLStatement_case()) {
+        case statementType::kCreateTable:
+            return toString(statement.createtable());
+        //TODO: add more cases here for additional APIs
+        default:
+            return "";
+    }
+}
+
+// generates a 'CREATE TABLE ...' statement
+std::string toString(CreateTable createTable) {
+    return absl::Substitute("CREATE TABLE $0 ([ $1,$2 ]) PRIMARY KEY ( [$3] )",
+                createTable.tablename(), 
+                toString(createTable.primarykeys()),
+                toString(createTable.nonprimarykeys()),
+                toPrimaryKeys(createTable.primarykeys()));
+}
+
+// converts a list of columns to a string format of {column1},{column2},...
+std::string toString(const google::protobuf::RepeatedPtrField<Column> columns) {
+    std::string columnsAsString = "";
+    for (int i = 0; i < columns.size(); i++) {
+        const Column& currCol = columns.at(i);
+        absl::StrAppend(&columnsAsString, toString(currCol));
+    }
+    // toString(currCol) will have an extra comma at the end so chop it off
+    columnsAsString = columnsAsString.substr(0, columnsAsString.size() - 1);
+    return columnsAsString;
+}
+
+// A column is { column_name data_type [ NOT NULL ] [ options_def ] }
+// where [ NOT NULL ] is optional and [ options_def ] is 
+// { OPTIONS ( allow_commit_timestamp = { true | null } ) }
+std::string toString(Column column) {
+    return absl::Substitute("{ $0 $1 $2 $3 },", 
```

Comment:
I believe in the actual DDL, the brackets should be removed.  Consider the create table statement from the test code:

CREATE TABLE Singers (
              SingerId   INT64 NOT NULL,
              FirstName  STRING(1024),
              LastName   STRING(1024),
              SingerInfo BYTES(MAX)
          ) PRIMARY KEY (SingerId))